data-prepper-plugins/aws-plugin/src/main/java/org/opensearch/dataprepper/plugins/aws/AwsSecretsSupplier.java

```diff
@@ -0,0 +1,99 @@
+package org.opensearch.dataprepper.plugins.aws;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.opensearch.dataprepper.aws.api.SecretsSupplier;
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueRequest;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueResponse;
+import software.amazon.awssdk.services.secretsmanager.model.ResourceNotFoundException;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class AwsSecretsSupplier implements SecretsSupplier {
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final TypeReference<Map<String, String>> MAP_TYPE_REFERENCE = new TypeReference<>() {
+    };
+
+    private final Map<String, Object> secretIdToValue;
+
+    public AwsSecretsSupplier(final AwsSecretPluginConfig awsSecretPluginConfig) {
+        secretIdToValue = toSecretMap(awsSecretPluginConfig);
+    }
+
+    private Map<String, Object> toSecretMap(final AwsSecretPluginConfig awsSecretPluginConfig) {
+        final Map<String, SecretsManagerClient> secretsManagerClientMap = toSecretsManagerClientMap(
+                awsSecretPluginConfig);
+        final Map<String, AwsSecretManagerConfiguration> awsSecretManagerConfigurationMap = awsSecretPluginConfig
+                .getAwsSecretManagerConfigurationMap();
+        return secretsManagerClientMap.entrySet().stream()
+                .collect(Collectors.toMap(Map.Entry::getKey, entry -> {
+                    final String secretConfigurationId = entry.getKey();
+                    final AwsSecretManagerConfiguration awsSecretManagerConfiguration =
+                            awsSecretManagerConfigurationMap.get(secretConfigurationId);
+                    final SecretsManagerClient secretsManagerClient = entry.getValue();
+                    final GetSecretValueRequest getSecretValueRequest = awsSecretManagerConfiguration
+                            .createGetSecretValueRequest();
+                    final GetSecretValueResponse getSecretValueResponse;
+                    try {
+                        getSecretValueResponse = secretsManagerClient.getSecretValue(getSecretValueRequest);
+                    } catch (Exception e) {
+                        throw ResourceNotFoundException.builder()
```

Comment:
Why are you throwing an AWS SDK exception here?

Reply:
I was trying to wrap it with a more explicit error message. Maybe RuntimeException is more appropriate?